ENH: Adding `order=` keyword to `np.eye()`.

dhermes opened the pull request:
Fixes #9995.

/cc @eric-wieser @njsmith

dhermes committed:
Adding `order=` keyword to `np.eye()`.

Fixes #9995.

dhermes commented on numpy/matlib.py:
```diff
@@ -210,7 +213,7 @@ def eye(n,M=None, k=0, dtype=float):
 
     """
-    return asmatrix(np.eye(n, M, k, dtype))
+    return asmatrix(np.eye(n, M=M, k=k, dtype=dtype, order=order))
```
Switching from positional to keyword is a bit of editorializing by me and I'm happy to revert if desired.

eric-wieser replied:
Keywords are slower, which is why you'll often see them omitted.

eric-wieser replied:
On the other hand, the more things to deter users from `matlib`, the better!

eric-wieser commented on numpy/tests/test_matlib.py:
```diff
@@ -28,10 +28,19 @@ def test_identity():
+    assert not xc.flags.f_contiguous
+
+    xf = numpy.matlib.eye(3, 4, dtype=int, order='F')
+    assert_array_equal(x, np.matrix([[ 1,  0,  0,  0],
```
NameError: global name 'x' is not defined

dhermes replied:
MERP! Copy-pasta (was in a hurry last night and didn't run tests locally, mea culpa).

eric-wieser commented:
Oh wow, this really is a simple fix!

seberg commented on numpy/lib/twodim_base.py:
```diff
@@ -153,6 +153,9 @@ def eye(N, M=None, k=0, dtype=float):
       Data-type of the returned array.
+    order : {'C', 'F'}, optional
+        Whether the output should be stored in row-major (C-style) or
+        column-major (Fortran-style) order in memory.
```
The versionadded :: tag would be good (grep the code for examples).

seberg replied:
Ah, also, please format the commit message according to our contribution guide, such as "ENH: ..."

seberg replied:
(Then again, I guess nowadays, we can also do that during merging in the squashing, didn't really get used to that yet, heh)

dhermes replied:
1. I said `1.13.4` but feel like that isn't quite the right version?
1. I just read the [commit guide][1], thanks!

[1]: https://docs.scipy.org/doc/numpy-dev/dev/gitwash/development_workflow.html#writing-the-commit-message

charris commented:
We should deprecate `matlib` at some point, it is all matrix based and we want to get rid of matrices.

dhermes committed:
Fix `x` -> `xf` variable name in `test_eye()`.

dhermes committed:
Adding `versionadded` Sphinx directive to newly added arguments.

seberg commented:
New features do not go into minor versions, so it will be 1.14 (unless
that would be already frozen, but I don't think it is).

dhermes commented:
WFM, before I make another commit, would you prefer `1.14` or `1.14.0` in the note? (I assume `1.14` but want to verify.)

seberg commented:
I am surprised, but it seems 1.14.0 is more common, both is being used
though, so I do no think it matters either way.

dhermes committed:
Changing `versionadded` to `1.14.0`.

(Also moving the directive to `eye()` from `empty()` in
`numpy.matlib`, it was there by mistake.)

dhermes commented:
@seberg Thanks! Sent a commit with `1.14.0` (also noticed I had accidentally put the `versionadded` on `numpy.matlib.empty`, d'oh!).

eric-wieser commented:
Thanks @dhermes for spotting this easy fix!
